api-gateway/src/main/java/org/swasth/apigateway/models/BaseRequest.java

```diff
@@ -92,9 +99,19 @@ protected void validateCondition(Boolean condition, ErrorCodes errorcode, String
     }
 
     protected void validateParticipant(Map<String,Object> details, ErrorCodes code, String participant) throws ClientException {
+        String participantCode = "1" + (String) details.get("participant_code");
+        ArrayList<String> roles = (ArrayList) details.get("roles");
+
         if(details.isEmpty()){
             throw new ClientException(code, participant + " does not exist in registry");
-        } else if(StringUtils.equals((String) details.get("status"), BLOCKED) || StringUtils.equals((String) details.get("status"), INACTIVE)){
+        }
+        if(participantCode.equals(hcxCode)) {
+            throw new ClientException(code, participant + " should not be sent as recipient in the incoming requests");
+        }
+        if(roles.contains(hcxRoles)) {
```

Comment:
hcxRoles is a single value or multiple. Are we handling the case sensitive input?

Reply:
single value